File: backendPidev/src/main/java/com/pidev/backend/Entity/Question.java
package com.pidev.backend.Entity;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "Question")
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private String id;
    private String contenue;
    private List<Technologie> tech;
    @DBRef

    private List<Hashtag> hashtag;
    @DBRef

    private List<Reponse> reponses;
    @DBRef

    private List<Vote> votes=new ArrayList<Vote>();
    @DBRef

    private User user;

}

File: Angular backOffice/src/app/model/question.ts
import { Hashtag } from "./hashtag";
import { Reponse } from "./reponse";
import { Technologie } from "./technologie";
import { User } from "./user";
import { Vote } from "./vote";

export class Question {
    id!:String;
    contenue!:String;
    tech!:Technologie[];
    hashtag!:Hashtag[];
    reponses:Reponse[]=[];
    votes!:Vote[];
    user!:User;


}